shared/modules/contract-utils.ts

```diff
@@ -1,25 +1,20 @@
 import pify from 'pify';
+import type EthQuery from '@metamask/eth-query';
 
 export type Contract = {
   contractCode: string | null;
   isContractAddress: boolean;
 };
 
-// Note(@dbrans): This is a simplified version of the 'EthQuery' interface specific to this file.
-type EthQueryWithGetCode = {
```

Comment:
Unfortunately, `getCode` is not included in type `EthQuery` so whenever we invoke `readAddressAsContract` while passing instance created by `new EthQuery`, this type check will fail. A key check is added `if (ethQuery && 'getCode' in ethQuery) {` to solve this issue, and we use `EthQuery` type everywhere